lib/tesla/middleware/logger.ex

```diff
@@ -176,6 +180,27 @@ defmodule Tesla.Middleware.Logger do
   config :tesla, Tesla.Middleware.Logger,
     filter_headers: ["authorization"]
   ```
+
+  ### Configure Logger metadata
+
+  Set `metadata: true` to include metadata in the log output.
+
+  ```
+  plug Tesla.Middleware.Logger, metadata: true
+  ```
+
+  Pass a list of atoms to `metadata` to include only specific metadata.
+
+  ```
+  plug Tesla.Middleware.Logger, metadata: [:url, :status, :request_body]
+  ```
+
+  Use `:conceal` request option to conceal sensitive requests.
+
+  ```
+  Tesla.get(client, opts: [conceal: true]])
```

Comment:
Overall, I like the idea, the only feedback is about documentation.

We should document `in one place` all the "reserved" options that have a given intent.

I like it!